Fix missing state class for reactive power sensors after #1417

N3rdix opened the pull request:
After explicitly switching to reactive power device class I missed the state_class also depending on this device_class (#1417).

Should fix https://github.com/lbbrhzn/ocpp/issues/1449 (the raised repair regarding the changed unit of measurement is expected nevertheless)

drc38 commented:
@N3rdix it'd be useful to include an assert in the tests to confirm the unit returned for this device class is what is expected.

N3rdix commented:
@drc38 although I added an assert in the test_init.py it does not feel to be the best place. As we do not have a test_sensor.py yet and I am not confident enough yet with pytest to build such a test from scratch easily I thought an experienced view might help.
Happy for any kind of help 😄

drc38 commented:
> @drc38 although I added an assert in the test_init.py it does not feel to be the best place. As we do not have a test_sensor.py yet and I am not confident enough yet with pytest to build such a test from scratch easily I thought an experienced view might help. Happy for any kind of help 😄

Best to put the assert statements here https://github.com/lbbrhzn/ocpp/blob/d351da0c7cc7f4a0dcc135e6ddffe2d9a5814b1b/tests/test_charge_point_v16.py#L270
Using mock config creates the integration's sensors so no need to do so manually. Hope that helps.

drc38 commented:
Also add the device class here https://github.com/lbbrhzn/ocpp/blob/d351da0c7cc7f4a0dcc135e6ddffe2d9a5814b1b/custom_components/ocpp/const.py#L127

N3rdix commented:
Hi @drc38,
thanks for your hints, that helped to understand the tests better 👍 

I've added the asserts and adjusted the mock data accordingly. Do you think this is enough/ok?

drc38 commented on tests/test_charge_point_v16.py:
```diff
@@ -821,7 +823,7 @@ async def send_meter_periodic_data(self):
                             "context": "Sample.Periodic",
                             "measurand": "Power.Reactive.Import",
-                            "unit": "W",
+                            "unit": "var",
```
@N3rdix looks good to me, you could remove this unit line which would confirm the default unit of var is being assigned by the integration.

N3rdix replied:
When I do that the asserts both fail but I could not yet find the reason yet (must be test related as the integration itself creates the HA entity correctly now in the devcontainer) 😕

drc38 replied:
Ahhh, that's because the metric class is not actually the sensor but a representation of the raw data read from the charger. To get the sensor value or unit for the assert it'd be similar to line 14 in this https://github.com/MatthewFlamm/pytest-homeassistant-custom-component/blob/master/tests/test_sensor.py

N3rdix replied:
That added the missing piece to my head 😄 I've added a separate `test_sensor.py`, I think this might be helpful for future sensor tests anyways

drc38 replied:
Just be aware that in order to test/assert in a sensor how data is processed you'd need to set the metric value/unit in the central system first as there is no dummy charger injecting data.
